backend/tournesol/serializers/contributor_recommendations.py

```diff
@@ -20,7 +21,9 @@ class Meta(RecommendationSerializer.Meta):
             - {"n_comparisons"}
             - {"n_contributors"}
             | {"is_public"}
+            | set(EntityNoExtraFieldSerializer.Meta.fields)
```

Comment:
Alright, I've checked the front end is this field is not required to display anything, but only to make the response of the contributor's recommendations API compliant with the type `RelatedEntityObject`.

Suggestion:

1. add the field `type` directly in the `RecommendationSerializer`
2. this will make all recommendations API compliant with the front end type `RelatedEntityObject`
3. delete the `| ContributorRecommendations` in the front end file `src/utils/types.ts`

and everything should be OK on the back end side : )

Reply:
The idea behind using `EntityNoExtraFieldSerializer` was that if new fields are added to it, and required by `VideoCard`, then we would not have to change this serializer too (and the same could be applied to all serializers that produce something that must be sent to a `VideoCard`). Kind of defining an interface for serializers whose result is used in video cards. Using `EntityNoExtraFieldSerializer` for that is probably not the best way though. I understand it adds complexity, but duplicating the required fields on each serializer doesn't seem ideal either.

I think it would be cleaner to have an extra `entity` field that would use a standard serializer that always works with video cards.

Anyway I applied your suggestions in 2db1bc40d957018f15333c5576cadef9b69c3bd1. I still had to make `metadata` read only to make it required.

I fail to understand why we can now remove `ContributorRecommendations` from `RelatedEntityObject` though. If it's because it now has all the same fields than `EntityNoExtraField` and `RelatedEntity` so typescript doesn't care, then I don't understand why I had to add it before (I also made it have all the same fields).

Reply:
> I understand it adds complexity, but duplicating the required fields on each serializer doesn't seem ideal either.

I think we are good, as the complexity added by the original proposition is far more important than configuring two serializers with the same required fields. This way they remain uncoupled.

> I think it would be cleaner to have an extra entity field that would use a standard serializer that always works with video cards.

Hmm, I'm not sure I understand, maybe we could talk about it quickly on Friday after the meeting?

> I fail to understand why we can now remove ContributorRecommendations from RelatedEntityObject though. If it's because it now has all the same fields than EntityNoExtraField and RelatedEntity so typescript doesn't care, then I don't understand why I had to add it before (I also made it have all the same fields).

You're right, you don't need `ContributorRecommendations` anymore in the front end because, by adding `type`, you made the entities returned by the recommendations API (both global and personal) compliant with what the front end is  expecting.

On `main` the page `page/videos/VideoRatings` only works with the rating API which has one subtle difference with the recommendations APIs: the presence of the field `type`. This difference was not intended, it's just we never have needed the field `type` until now.

The back end should as much as possible returns the same representation for an entity, regardless of the endpoint. These difference with `type` could be seen as a design bug. Maybe the `RecommendationsSerializer` should inherit its "base" fields from the `EntitySerializer`, and maybe all endpoints dealing with entities should be based on `EntitySerializer` to avoid having the same error in the future.

Reply:
> This way they remain uncoupled.

I agree my proposition was not good but I still think all the serializers whose output is used in video cards *should* be coupled somehow. If we change the video card to display a new entity attribute for example, it would be better not to have to change multiple serializers to add this field.

> > I think it would be cleaner to have an extra entity field that would use a standard serializer that always works with video cards.
> Hmm, I'm not sure I understand, maybe we could talk about it quickly on Friday after the meeting?

Sure.

My suggestion is that the API would work like `users_me_contributor_ratings_list`: it would return something with an `entity` field, instead of returning an entity with additional fields. That way we could use a generic serializer to produce the `entity` field (like `ContributorRatingSerializer` uses `EntityNoExtraFieldSerializer` to serialize the entity).

> The back end should as much as possible returns the same representation for an entity, regardless of the endpoint. 

Indeed, that's the reason for my points above.

> Maybe the RecommendationsSerializer should inherit its "base" fields from the EntitySerializer, and maybe all endpoints dealing with entities should be based on EntitySerializer to avoid having the same error in the future.

Yes that would work too to avoid having to change multiple serializers when we want to change how an entity is serialized. I think having the entities always serialized the same way (or a few ways) would be slightly cleaner though (compared to having multiple API returning entities with some common fields and some different extra fields). But I'm not sure which one is preferable in practice.

Reply:
These changes can be discussed or done in a separate issue or PR.

Reply:
> These changes can be discussed or done in a separate issue or PR.

Yes sure, it's really out of the initial scope : )